joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/parser/RubyJsonHelpers.scala

```diff
@@ -137,19 +152,85 @@ object RubyJsonHelpers {
       }
     }
 
+    /** Attempts to evaluate and parse the collection associated with the splattingField, generating FieldDeclarations
+      * for each of the elements.
+      * @param fieldStmts
+      *   List of all the field statements
+      * @param splattingFields
+      *   List of splatting fields
+      * @return
+      *   List of:
+      *   - Some(_) => if splattingField either is evaluated to a list of FieldDeclarations, otherwise a SimpleCall
+      *   - None => if splattingField cannot be evaluated to either FieldsDeclaration or SimpleCall
+      */
+    def lowerSplattingFieldDecl(
```

Comment:
Is it not possible to move this lowering closer to AstCreator?

Reply:
I can't think of a way. This determines whether something is a FieldsDeclaration at the top of a class, or whether it is a SimpleCall that's going in the `<body>` method. I can't see a way that we have this somewhere in the AstCreator and make it work without making it extremely convoluted

Reply:
Yeah that's fair. I'm keen to get this in so the issue wil be fixed then we can revisit the location of the lowering